starter_kits/C/src/MyBot.c

```diff
@@ -0,0 +1,324 @@
+// First off, here are some defines and includes. I forget what exactly needs what
+#include <stdio.h>
+#include <stdlib.h>
+
+// We include time.h so we can seed the RNG for default "smart" movement with time()
+#include <time.h>
+
+
+
+// Now for our data structures. These are intentionally left rather basic, so add whatever you want to them
+struct ship{
+	unsigned char x, y;
+	unsigned short halite;
+} ships[4096];
+
+struct player{
+	unsigned char dropoffs[64], dropoff_count, ship_count, ship_yard[2];
+	unsigned short ships[1024];
+	unsigned int halite;
+} players[4];
+
+struct map{
+	unsigned short halite;
+	unsigned short occupied; //(unsigned short)-1 means it's not occupied
+} map[64][64];
+
+
+
+// Now for some global variables to help us keep track of things
+unsigned char dropoffs[256][2], extract_ratio, inspiration_enabled, inspiration_radius, inspiration_ship_count, inspired_extract_ratio, inspired_bonus_multiplier, inspired_move_cost_ratio, mhh, mhw, move_cost_ratio, my_id, map_size[2], number_of_players;
+unsigned short dropoff_cost, max_energy, max_turns, ship_cost, turn_number;
+
+FILE *log_file = NULL;
+
+
+
+// Now let's declare some helper functions
+// You might reasonably want to keep some of these
+
+// Normalize X value (map_width + 1 = 0, 0 - 1 = map_width)
+inline unsigned char norm_x(char x){
```

Comment:
`inline`-ing these functions is a bit overkill but I know some people want the performance

Reply:
Is there a reason the functions shouldn't be inlined? They were originally macros so they'd be inlined, but snaar mentioned that macros are old and bad, so I changed them to inline functions

Reply:
inline is fine here and makes sense, just unusual and a micro optimization (they usually cause trouble)